cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -727,17 +727,14 @@ std::vector<size_t> reader::impl::calculate_output_num_rows_per_source(size_t co
   // Binary search start_row and end_row in exclusive_sum_num_rows_per_source vector
   auto const start_iter =
     std::upper_bound(partial_sum_nrows_source.cbegin(), partial_sum_nrows_source.cend(), start_row);
-  auto const end_iter =
-    (end_row == _file_itm_data.global_skip_rows + _file_itm_data.global_num_rows)
-      ? partial_sum_nrows_source.cend() - 1
-      : std::upper_bound(start_iter, partial_sum_nrows_source.cend(), end_row);
+  auto const end_iter = std::lower_bound(start_iter, partial_sum_nrows_source.cend(), end_row);
 
   // Compute the array offset index for both iterators
   auto const start_idx = std::distance(partial_sum_nrows_source.cbegin(), start_iter);
   auto const end_idx   = std::distance(partial_sum_nrows_source.cbegin(), end_iter);
-
-  CUDF_EXPECTS(start_idx <= end_idx,
-               "Encountered invalid source files indexes for output chunk row bounds");
+  auto const max_idx   = static_cast<cudf::size_type>(partial_sum_nrows_source.size());
```

Comment:
I’d like an indication that this is a “size” as opposed to a computed maximum. Maybe name it `num_sources`?

Reply:
All done!